### PR TITLE
rsx/overlays: Misc Improvements

### PR DESCRIPTION
**Pull request by kd-11**
- Unify overlay rendering shaders to reduce the headache of maintaining 2 separate paths
- Add support for disabling vertex snap on the overlays. This fixes rendering the round parts of rounded rectangles.
- Fix grid snap code which was broken for years due to a misplaced bracket 🙈 
- Force disable the rounded rectangles feature on mac OS which doesn't support triangle fans. A workaround will be provided at a later time.

### CODE REVIEW

File: rpcs3/Emu/RSX/Overlays/overlay_controls.cpp [hunk 1]
```diff
@@ -940,7 +940,11 @@ namespace rsx
 			{
 				compiled_resources.clear();
 
-				if (radius == 0 || radius > (w / 2))
+				if (radius == 0 ||
+#ifdef __APPLE__
+					true ||
+#endif
+					radius > (w / 2))
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/Overlays/overlay_controls.cpp:947]: you can just ifdef the whole if instead of doing this split

**Comment by Megamouse** [comment]
hmm maybe also add a comment. otherwise noone knows why it's ifdefed

**Comment by kd-11** [comment]
> hmm maybe also add a comment. otherwise noone knows why it's ifdefed

I think the `__APPLE__` part says it all :laughing:
It's temporary anyway, I'll get to it when I can build macos without tearing my hair out.

**Comment by Megamouse** [comment]
maybe show comparisons. too lazy to take a look lmao

**Comment by kd-11** [comment]
After:
image

**Pull request merged**